firefox_puppeteer/tests/examples/test_marionette_intro.py

```diff
@@ -0,0 +1,148 @@
+        #     """, script_args=[url])
+
+        # # Let's try our navigate method.
+        # destination_url = "about:blank"
+        # my_navigate(destination_url)
+        # self.assertEqual(self.marionette.get_url(), destination_url)
+
+        # # Ok, let's see what happens when we hit the network:
+        # destination_url = "http://people.mozilla.org/~cmanchester"
+        # my_navigate(destination_url) # self.marionette.navigate(destination_url)
+        # self.assertEqual(self.marionette.get_url(), destination_url)
+        pass
+
+    def test_element(self):
+        # This test works with the urlbar element, so unlike tests in the rest
+        # of the file, we're going to run this test in chrome context. Because
+        # we set context to content in setUp, the rest of the tests methods
+        # will not be harmed by this, even if we do nothing at the end of the
+        # method.
+        self.marionette.set_context("chrome")
+
+        # This test needs to interact with the awesomebar. The awesomebar is
+        # just a xul element somewhere we need to select...
+
+        # Getting a handle to an element is a matter of calling "find_element".
+        # The first argument is a strategy (one of some constants defined in
+        # Marionette.By, which are just strings). The second argument is a
+        # "target" whose meaning depends on the strategy. A common strategy is
+        # to retrieve an element by its id.
+
+        # # How can we find the back button?
+        # back_button = self.marionette.find_element(By.ID, "back-button")
+
+        # If we had guessed wrong, we would have run into problems:
+        # back_button = self.marionette.find_element(By.ID, "backbutton")
+
+        # # Ok, we got lucky! What about the urlbar?
+        # urlbar = self.marionette.find_element(By.ID, "urlbar")
+
+        # # Once we have an element, we can simulate typical input interactions
+        # # and inspect arbitrary properties. For instance:
+        # print "Is the urlbar focused before our click? %s" % urlbar.get_attribute('focused')
+        # urlbar.click()
+        # print "Is the urlbar focused after our click? %s" % urlbar.get_attribute('focused')
+        # data_uri = 'data:text/html,<h1>Marionette sent these keys</h1>'
+        # urlbar.send_keys(data_uri)
+        # print "The value of the urlbar is: %s" % urlbar.get_attribute('value')
+        # urlbar.send_keys(Keys.ENTER)
+        # with self.marionette.using_context("content"):
+        #     current_url = self.marionette.get_url()
+        #     print "The current url is: %s" % current_url
+
+        # # As above, let's see what happens when we hit the network:
+        # network_uri = "http://people.mozilla.org/~cmanchester"
+        # urlbar.send_keys(network_uri + Keys.ENTER)
+        # with self.marionette.using_context("content"):
+        #     current_url = self.marionette.get_url()
+        #     print "The current url is: %s" % current_url
```

Comment:
I would suggest we also have an example which shows the same steps but with our own puppeteer library. It would be good to see for people how easier things are getting.